src/sif/dynamiccost.cc

```diff
@@ -74,8 +74,9 @@ DynamicCost::DynamicCost(const CostingOptions& options,
                          const TravelMode mode,
                          uint32_t access_mask,
                          bool penalize_uturns)
-    : pass_(0), allow_transit_connections_(false), allow_destination_only_(true), travel_mode_(mode),
-      access_mask_(access_mask), closure_factor_(kDefaultClosureFactor), flow_mask_(kDefaultFlowMask),
+    : pass_(0), allow_transit_connections_(false), allow_destination_only_(true),
+      allow_conditional_destination_(false), travel_mode_(mode), access_mask_(access_mask),
```

Comment:
I'm a bit concerned that we have different defaults for regular and conditional 'destination' behavior. In what cases are the defaults used? (like I know `allow_destination_only_` is explicitly overridden in bidir a*)

Reply:
I can say only about `allow_conditional_destination_`.  
Consider next:
1. The first attempt of finding the shortest path: it does not include edges in the middle of the path with active conditional destination restrictions -> the flag allow_conditional_destination_ is false
2. If first attempt is failed. (A path was not found) the second attempt of finding the shortest path is performed. We need relax this (set allow_conditional_destination_ to true) and we allow to go through edges with these active restrictions. 

I think the same as you about allow_destination_only_. This is overridden only for bidir a*